Makefile

```diff
@@ -278,8 +282,11 @@ $(ISO): $(ALL)
 VAGRANT ?= vagrant
 VAGRANT_DEFAULT_PROVIDER ?= virtualbox
 VAGRANT_UP_OPTS ?= --provision --no-destroy-on-error --parallel --provider=$(VAGRANT_DEFAULT_PROVIDER)
+VAGRANT_PRESHARED_KEY_PATH ?= .vagrant/preshared_key_for_k8s_nodes
 
 vagrantup: $(ALL) ## Run 'vagrant up' to (re-)provision a development environment
+	mkdir -p .vagrant/
+	if [ ! -f $(VAGRANT_PRESHARED_KEY_PATH) ]; then ssh-keygen -t rsa -b 4096 -N '' -f $(VAGRANT_PRESHARED_KEY_PATH) -C Makefile ; fi
```

Comment:
Maybe you could have this key define as an independent target, e.g.:

```
vagrantup: $(ALL) $(VAGRANT_PRESHARED_KEY)
    $(VAGRANT) up $(VAGRANT_UP_OPTS)

$(VAGRANT_PRESHARED_KEY):
    mkdir -p $(@D)
    ssh-keygen -t rsa -b 4096 -N '' -f $@ -C Makefile
```

Reply:
Done. Thx for the tip.